resources/ts/Components/IntegrationCard.tsx

```diff
@@ -38,17 +41,69 @@ export const OpenWidgetBuilderButton = ({
   }
 
   return (
-    <ButtonLinkSecondary href={`/integrations/${id}/widget`} target={"_blank"}>
+    <ButtonLinkSecondary
+      href={`/integrations/${id}/widget`}
+      target="_blank"
+      className="flex self-start"
+    >
       {t("integrations.open_widget")}
     </ButtonLinkSecondary>
   );
 };
 
-export const IntegrationCard = ({ id, name, type, status, onEdit }: Props) => {
+export const IntegrationCard = ({
+  id,
+  name,
+  type,
+  status,
+  credentials,
+  onEdit,
+}: Props) => {
   const { t } = useTranslation();
 
   const integrationTypesInfo = useIntegrationTypesInfo();
 
+  const auth0TestClient = credentials.auth0.find(
+    (client) =>
+      client.integrationId === id && client.tenant === Auth0Tenant.Testing
+  );
+  const auth0ProdClient = credentials.auth0.find(
+    (client) =>
+      client.integrationId === id && client.tenant === Auth0Tenant.Production
+  );
+  const uiTiDv1TestConsumer = credentials.uitidV1.find(
+    (client) =>
+      client.integrationId === id &&
+      client.environment === UiTiDv1Environment.Testing
+  );
+  const uiTiDv1ProdConsumer = credentials.uitidV1.find(
+    (client) =>
+      client.integrationId === id &&
```

Comment:
I would perhaps only pass the credentials of the given integration to the integration card, it would make all those simpler and would mean the card doesn't have to worry about receiving another integration's data